.github/workflows/Pre_release.yml

```diff
@@ -12,29 +20,25 @@ jobs:
   release:
     name: "Publish Pre-Release"
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
     steps:
       - uses: actions/checkout@v4
         with:
-          fetch-depth: 0  # Needed determine latest tag
+          fetch-depth: 0  # Needed determine tags
           persist-credentials: true
-          ref: main
-      - name: Get Latest Tag
-        id: latest_tag
-        uses: WyriHaximus/github-action-get-previous-tag@v1
-        # TODO: in future we might need to perform backports, at which point we'll need to check
-        # that the project.version has not already been tagged 
-      - name: Check for updated Project version
+          ref: ${{ inputs.commit }}
+      - name: Check for new Project version
         id: project_toml
         shell: julia --color=yes --project {0}
         env:
           LATEST_TAG: ${{ steps.latest_tag.outputs.tag }}
         run: |
           using Pkg.Types
           project = read_project("Project.toml")
-          latest_tag = parse(VersionNumber, ENV["LATEST_TAG"])
+          tags = parse.(VersionNumber, split(readchomp(`git tag -l`), "\n"))
           @info "Project version: project.version"
-          @info "Latest tag: latest_tag"
-          project.version > latest_tag || exit(1)  # error if can't make a tag
+          project.version in tags && error("Tag v$(project.version) already exists")
```

Comment:
Doesn't `project.version` lack the `v` suffix while `tags` would include it?

Reply:
no it get's parsed as a `VersionNumber` 
```julia
julia> project.version
v"0.0.4-dev"
```

Reply:
TIL:
```julia
julia> parse(VersionNumber, "v0.0.1")
v"0.0.1"

julia> parse(VersionNumber, "0.0.1")
v"0.0.1"
```